source/Calamari.Tests/KubernetesFixtures/KubernetesContextScriptWrapperLiveFixtureBase.cs

```diff
@@ -214,34 +212,22 @@ protected void DoDiscovery(AwsAuthenticationDetails authenticationDetails)
                 properties);
 
             Log.ServiceMessages.Should()
-                .ContainSingle(s => s.Properties["name"] == properties["name"])
+                .ContainSingle(s =>
+                    s.Name == KubernetesDiscoveryCommand.CreateKubernetesTargetServiceMessageName &&
+                    s.Properties["name"] == properties["name"])
                 .Which.Should()
                 .BeEquivalentTo(expectedServiceMessage);
         }
 
-        protected CalamariResult ExecuteDiscoveryCommand<TAuthenticationDetails>(
-            TargetDiscoveryContext<TAuthenticationDetails> discoveryContext,
-            IEnumerable<string> extensions,
-            params (string key, string value)[] otherVariables)
+        protected void ExecuteDiscoveryCommandAndVerifyResult<TAuthenticationDetails>(
+            TargetDiscoveryContext<TAuthenticationDetails> discoveryContext)
             where TAuthenticationDetails : class, ITargetDiscoveryAuthenticationDetails
         {
-            using (var variablesFile = new TemporaryFile(Path.GetTempFileName()))
-            {
-                variables.Add(KubernetesDiscoveryCommand.ContextVariableName, JsonConvert.SerializeObject(discoveryContext));
-                foreach (var (key, value) in otherVariables)
-                    variables.Add(key, value);
-
-                variables.Save(variablesFile.FilePath);
-
-                var result = InvokeInProcess(Calamari()
-                       .Action(KubernetesDiscoveryCommand.Name)
-                       .Argument("variables", variablesFile.FilePath)
-                       .Argument("extensions", string.Join(',', extensions)));
+            variables.Add(KubernetesDiscoveryCommand.ContextVariableName,
+                JsonConvert.SerializeObject(discoveryContext));
 
-                WriteLogMessagesToTestOutput();
-
-                return result;
-            }
+            SetupTempDirectoryAndVerifyResult(null,
+                (dir, _) => ExecuteCommand(KubernetesDiscoveryCommand.Name, dir, null), shouldSucceed: true);
```

Comment:
I've updated target discovery tests to execute out-of-process too so that all kubernetes tests are consistent.